.github/workflows/merge-commits-to-next-branch.yml

```
@@ -0,0 +1,134 @@
+name: Merge commits to next branch
+
+on:
+  schedule:
+    - cron: "0 */2 * * *"
+
+env:
+  COMMIT_INFO: commit-info
+
+jobs:
+  dequeue:
+    runs-on: ubuntu-latest
+    steps:
+      steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - run: npm ci
+      - run: npm install -g @azure/storage-queue
+      - uses: actions/github-script@v6
+        with:
+          script: |
+            const { QueueClient, QueueServiceClient } = require('@azure/storage-queue')
+            const connectionString = `${{ secrets.AZURE_STORAGE_CONNECTION_STRING }}`;
+            const queueName = `${{ secrets.AZURE_QUEUE_NAME }}`;
+            const queueServiceClient = QueueServiceClient.fromConnectionString(connectionString);
+            const queueClient = queueServiceClient.getQueueClient(queueName);
+            var receivedMessages = await queueClient.receiveMessages();
+            const message = receivedMessages.receivedMessageItems[0];
+            $COMMIT_INFO = JSON.parse(message.messageText);
+  create-branch:
+    runs-on: ubuntu-latest
+    if: ( $COMMIT_INFO.label == 'queued' )
+    steps:
+      - name: Create branch
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        with:
+          script: |
+            git fetch origin
+            git checkout test_next
+            git checkout -b main-next-$COMMIT_INFO.sha
+            git merge $COMMIT_INFO.sha --no-ff
+  reset-branch:
+    runs-on: ubuntu-latest
+    if: ( $COMMIT_INFO.label == 'queued' && needs.create-branch.result == 'failure'
+      )
+    steps:
+      - name: Reset main to a commit
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        with:
+          script: |
+            git fetch origin
+            git checkout main
+            git branch -D main-next-$COMMIT_INFO.sha
+            git checkout -b main-next-$COMMIT_INFO.sha
+            git reset --hard $COMMIT_INFO.sha
```

Are you trying to reset main to a commit, or reset the `main-next-COMMIT_SHA` branch? As it is, you're resetting the `main-next-COMMIT_SHA` branch to `$COMMIT_INFO.sha`. If that's what you want, could you simplify like this?

```suggestion
            git fetch origin
            git checkout main-next-$COMMIT_INFO.sha
            git reset --hard $COMMIT_INFO.sha
```

---

yep -- I was thinking to resetting to a particular commit and then opening the PR against next. I'll simply this